src/main.rs

```diff
@@ -126,3 +131,38 @@ fn run_agent() -> failure::Fallible<()> {
 
     Ok(())
 }
+
+/// Deadend subcommand entry point.
+fn write_deadend_release_info(reason: Option<String>) -> failure::Fallible<()> {
+    if reason.is_some() {
+        // Avoid showing partially-written messages using tempfile and
+        // persist (rename).
+        let mut f = tempfile::Builder::new()
+            .prefix(".deadend.")
+            .suffix(".motd.partial")
+            // Create the tempfile in the same directory as the final MOTD,
+            // to ensure proper SELinux labels are applied to the tempfile
+            // before renaming.
+            .tempfile_in("/run/motd.d")
+            .with_context(|e| format!("failed to create temporary MOTD file: {}", e))?;
+        // Set correct permissions of the temporary file, before moving to
+        // the destination (`tempfile` creates files with mode 0600).
+        fs::set_permissions(f.path(), Permissions::from_mode(0o664))
+            .with_context(|e| format!("failed to set permissions of temporary MOTD file: {}", e))?;
+
+        if let Some(reason) = reason {
+            writeln!(
+                f,
+                "This release is a dead-end and won't auto-update: {}",
+                reason
+            )
+            .with_context(|e| format!("failed to write MOTD: {}", e))?;
+        }
+
+        f.persist("/run/motd.d/85-zincati-deadend.motd")
+            .with_context(|e| format!("failed to persist temporary MOTD file: {}", e))?;
+    } else {
+        std::fs::remove_file("/run/motd.d/85-zincati-deadend.motd")?;
```

Comment:
I think you'll need to special case the file not found case here, which is not an error condition in this case.